dvc/repo/experiments/remove.py

```diff
@@ -61,3 +65,16 @@ def _get_ref(ref_infos, name, cur_rev):
         msg.extend([f"\t{info}" for info in ref_infos])
         raise InvalidArgumentError("\n".join(msg))
     return ref_infos[0]
+
+
+def _remove_exp_by_name(repo, exp_name: str):
+    ref_info = _get_exp_ref(repo, exp_name)
+    if ref_info is not None:
+        remove_exp_refs(repo.scm, [ref_info])
+    else:
+        stash_index = _get_exp_stash_index(repo, exp_name)
+        if stash_index is None:
+            raise InvalidArgumentError(
+                f"'{exp_name}' is not a valid experiment name"
+            )
```

Comment:
We need to account for whether or not `exp_name` is the SHA for a stash revision here